tests/postgresql_test.go

```diff
@@ -0,0 +1,115 @@
+// +build integration
+
+package tests
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tidwall/gjson"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+const (
+	containerName = "nri-postgresql"
+	schema        = "jsonschema.json"
+)
+
+func executeDockerCompose(containerName string, envVars []string) (string, string, error) {
+	cmdLine := []string{"run"}
+	for i := range envVars {
+		cmdLine = append(cmdLine, "-e")
+		cmdLine = append(cmdLine, envVars[i])
+	}
+	cmdLine = append(cmdLine, containerName)
+	fmt.Printf("executing: docker-compose %s\n", strings.Join(cmdLine, " "))
+	cmd := exec.Command("docker-compose", cmdLine...)
+	var outbuf, errbuf bytes.Buffer
+	cmd.Stdout = &outbuf
+	cmd.Stderr = &errbuf
+	err := cmd.Run()
+	stdout := outbuf.String()
+	stderr := errbuf.String()
+	return stdout, stderr, err
+}
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	result := m.Run()
+	os.Exit(result)
+}
+
+func TestSuccessConnection(t *testing.T) {
+	hostname := "db"
+	envVars := []string{
+		fmt.Sprintf("HOSTNAME=%s", hostname),
+		"USERNAME=postgres",
+		"PASSWORD=example",
+		"DATABASE=demo",
+	}
+	stdout, _, err := executeDockerCompose(containerName, envVars)
+	assert.Nil(t, err)
+	assert.NotEmpty(t, stdout)
+	response := string(stdout)
+	/**
+	err=validateJSONSchema(schema, response)
+	if err!=nil{
+		fmt.Println(err.Error())
+	}
+	assert.Nil(t, err)
+	 */
```

Comment:
why not using the schema validation here?

Reply:
That's weird, but the schema validation is successfully locally but not in the GH nodes...That shouldn't depend on the environment where the tests are executed...

Reply:
agree, let's troubleshoot the failure. I think keeping the schema validation make sense

Reply:
It's fixed now, the issue was that when running `docker-compose run` through the `os.Exec` the stdout contained more info than the JSON response (downloading docker images info...)  Now, I added docker-compose pull before running the docker-compose run and the JSON schema validation works as expected